checks/tasks/tls.py

```diff
@@ -2965,38 +2964,25 @@ def http_checks(af_ip_pair, url, task):
 def forced_http_check(af_ip_pair, url, task):
     """
     Check if the webserver is properly configured with HTTPS redirection.
-
     """
     # First connect on port 80 and see if we get refused
     try:
-        has_443 = False
-        conn, res, headers, visited_hosts = http_fetch(
-            url, af=af_ip_pair[0], path="", port=443, task=task, ip_address=af_ip_pair[1], depth=MAX_REDIRECT_DEPTH
-        )
-        has_443 = True
-        conn, res, headers, visited_hosts = http_fetch(
-            url, af=af_ip_pair[0], path="", port=80, task=task, ip_address=af_ip_pair[1], depth=MAX_REDIRECT_DEPTH - 1
-        )
-    except (OSError, http.client.BadStatusLine, NoIpError, ConnectionHandshakeException, ConnectionSocketException):
-        if has_443:
-            # If we got refused on port 80 the first time
-            # return the FORCED_HTTPS_NO_HTTP status and score
-            return scoring.WEB_TLS_FORCED_HTTPS_NO_HTTP, ForcedHttpsStatus.no_http
-        else:
-            # No connection anywhere; return failure.
-            return scoring.WEB_TLS_FORCED_HTTPS_BAD, ForcedHttpsStatus.bad
+        http_get_ip(hostname=url, ip=af_ip_pair[1], port=443, https=True)
+    except requests.RequestException:
+        # No HTTPS connection available
+        return scoring.WEB_TLS_FORCED_HTTPS_BAD, ForcedHttpsStatus.bad
+
+    try:
+        response_http = http_get_ip(hostname=url, ip=af_ip_pair[1], port=80, https=False)
+    except requests.RequestException:
+        # No plain HTTP available, but HTTPS is
+        return scoring.WEB_TLS_FORCED_HTTPS_NO_HTTP, ForcedHttpsStatus.no_http
 
-    # Valid if same domain, or *higher* domain. Use case:
-    # www.example.com:80 -> example.com:443. Example.com:443 can set HSTS
-    # with includeSubdomains
     forced_https = ForcedHttpsStatus.bad
     forced_https_score = scoring.WEB_TLS_FORCED_HTTPS_BAD
 
-    if 443 in visited_hosts and conn.port == 443:
-        for visited_host in visited_hosts[443]:
-            if visited_host in url:
```

Comment:
The old code seems to only accept the redirect if the redirect target is a substring of the domain under test. This may make sense for HSTS (see removed comment which seems entirely out of place here), but not for HTTPS redirect?

Reply:
This is intentional and should be kept in the test, per comment:
```
In case of redirecting, a domain should firstly upgrade itself by redirecting to its HTTPS version before it may redirect to another domain. This also ensures that the HSTS policy will be accepted by the web browser.
```